src/gov/nist/javax/sip/stack/SIPTransactionStack.java

```diff
@@ -544,6 +544,9 @@ protected SIPTransactionStack() {
         // The default (identity) address lookup scheme
 
         this.addressResolver = new DefaultAddressResolver();
+        
+        // Init vavles list
+        this.sipMessageValves = new ArrayList<SIPMessageValve>();
```

Comment:
as you suggested, "use an ordered-wise colelction like List" and Arraylist is keeping order of the list, come back to double quoted string PR, you have mentions using ArrayList so that I used ArrayList here. Do you have any orther Idea, I will check and update.